rust/extractor/src/main.rs

```diff
@@ -36,10 +51,16 @@ fn main() -> anyhow::Result<()> {
         with_proc_macro_server: ProcMacroServerChoice::Sysroot,
         prefill_caches: false,
     };
-    for input in cfg.inputs {
-        let (db, vfs, _macro_server) = load_workspace_at(&input, &config, &load_config, &progress)
-            .context("loading inputs")?;
-        _macro_server.expect("no macro server");
+    let projects = find_project_manifests(&cfg.inputs).context("loading inputs")?;
+    for project in projects {
+        let mut workspace = ProjectWorkspace::load(project, &config, &progress)?;
+
+        if load_config.load_out_dirs_from_check {
+            let build_scripts = workspace.run_build_scripts(&config, &progress)?;
+            workspace.set_build_scripts(build_scripts)
+        }
+
+        let (db, vfs, _macro_server) = load_workspace(workspace, &config.extra_env, &load_config)?;
```

Comment:
it's slightly weird to do an `if` on something hard-coded to `true `just 8 lines above this one. I understand you took this from `load_workspace_at` though. And, it does make it more robust if we ever tweak the configurations. Though to avoid the duplication from rust-analyzer code, maybe we can just call `load_workspace_at` on the manifests?
```suggestion
        let (db, vfs, _macro_server) = load_workspace_at(project, &config.extra_env, &load_config)?;
```
This will make a bit of unnecessary churning (rediscover the manifest itself), but it means we don't need to replicate `load_workspace_at` functionality in our code.

Reply:
Good point, thanks!